data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -135,8 +136,10 @@ protected long pos(T record) {
       Iterable<CloseableIterable<Record>> deleteRecords = Iterables.transform(deletes,
           delete -> openDeletes(delete, deleteSchema));
       StructLikeSet deleteSet = Deletes.toEqualitySet(
-          // copy the delete records because they will be held in a set
-          CloseableIterable.transform(CloseableIterable.concat(deleteRecords), Record::copy),
+          CloseableIterable.transform(
+              CloseableIterable.concat(deleteRecords, ThreadPools.getWorkerPool(), ThreadPools.getPoolParallelism()),
```

Comment:
did this PR actually change or increase the parallelism? it seems to me we are still using the default parallelism of `2 * ThreadPools.WORKER_THREAD_POOL_SIZE`.  Is the purpose of this PR mainly refactoring the API and supporting different parallelism (if needed in the future)?

Reply:
The purpose of this PR is to be able to read deleteFile in parallel, before this PR, iceberg will read all delete files in one thread  serially. Now, delete files will be read in parallel by a worker pool.